caffe2/mkl/operators/pool_op.cc

```diff
@@ -0,0 +1,196 @@
+#include "caffe2/core/context.h"
+#include "caffe2/core/operator.h"
+#include "caffe2/operators/conv_pool_op_base.h"
+
+#include "caffe2/utils/mkl_utils.h"
+
+#ifdef CAFFE2_HAS_MKL_DNN
+
+namespace caffe2 {
+
+namespace {
+// These two classe are just used as template arguments passed to the PoolOp
+// template to instantiate the different algorithms.
+class AveragePool {};
+class MaxPool {};
+}  // namespace
+
+
+namespace mkl {
+
+template <typename T, typename PoolType>
+class MKLPoolOp final : public ConvPoolOpBase<MKLContext>{
+
+  public:
+    USE_CONV_POOL_BASE_FUNCTIONS(MKLContext);
+    
+    MKLPoolOp(const OperatorDef &operator_def, Workspace *ws)
+        : ConvPoolOpBase<MKLContext>(operator_def, ws) {
+
+      CAFFE_ENFORCE(
+            dilation_h_ == 1 && dilation_w_ == 1,
+            "Pooling op does not support dilation right now.");
+        if (!global_pooling_) {
+      CAFFE_ENFORCE(
+          pad_t_ < kernel_h_ && pad_b_ < kernel_h_ && pad_l_ < kernel_w_ &&
+              pad_r_ < kernel_w_,
+          "Pad should be smaller than kernel.");
+        }
+    }
+    
+    ~MKLPoolOp() {
+        if (workspace_buffer_ != NULL) {
```

Comment:
Yangqing, can you please help me figure out which linter should I use. I use visual studio code for editing and it has inbuilt pylint for python but don't have any for c++. when I googled I find many c++ linters. can you please suggest which one should I go for?

Thanks!